rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -1321,6 +1321,52 @@ def test_upgrade_db_22_to_23_without_frontend_settings(data_dir, user_data_dir):
     assert db.get_version() == 23
 
 
+def test_upgrade_db_23_to_24(user_data_dir):  # pylint: disable=unused-argument
+    """Test upgrading the DB from version 23 to version 24.
+
+    Deletes the AdEx used query ranges, drops the AdEx events table and re-creates it.
+    """
+    msg_aggregator = MessagesAggregator()
+    _use_prepared_db(user_data_dir, 'v23_rotkehlchen.db')
+    db = _init_db_with_target_version(
+        target_version=24,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    cursor = db.conn.cursor()
+    # Make sure AdEx used query ranges have been deleted
+    assert cursor.execute(
+        'SELECT COUNT(*) from used_query_ranges WHERE name = "adex_events";',
+    ).fetchone()[0] == 0
+    # Make sure `adex_events` has been created
+    assert cursor.execute('SELECT COUNT(*) from adex_events;').fetchone()[0] == 0
+    query = cursor.execute(
```

Comment:
What are you doing here? You just checked this with a short line above. Why are you making another query specifically for every column to again make sure it's 0?

If it's to test that log_index can be queried and the table is upgrades just ask for that column alone ...

Reply:
I used the v21 to v22 test as an example of a migration that involved recreating a table. I thought that typing all the columns was a nice way of double checking the new schema and and migration was OK. I'll amend it asking the column alone.

Reply:
Hmmm okay then leave it as is. But write that as a comment on the top comment, to explain why this is being done.

This: 

>  thought that typing all the columns was a nice way of double checking the new schema and and migration was OK

Reply:
But the for loop underneath in the query is not needed. Probably can be done better. All you need to see is that this has nothing to return. Probably something like `len(query.fetchall()) == 0` ?